src/utils.jl

```diff
@@ -0,0 +1,82 @@
+"""
+    indices_of_unique(x) -> Dict
+
+Return a `Dict` whose keys are the unique elements of `x` and whose values are the
+corresponding indices in `x`.
+"""
+function indices_of_unique(x)
+    d = Dict{eltype(x),Vector{Int}}()
+    for (i, xi) in enumerate(x)
+        if haskey(d, xi)
+            push!(d[xi], i)
+        else
+            d[xi] = [i]
+        end
+    end
+    return d
+end
+
+"""
+    split_chain_indices(
```

Comment:
Isn't there some existing splitting functionality for `ess`? Is the plan to merge these eventually?

Reply:
Not quite merge, because there are two different types of splitting we can consider. This approach supports ragged chains and is as a result more complex and doesn't discard any draws (instead dividing the remainder across the earlier splits).

For `ess`/`rhat`, we don't support ragged chains so would discard draws if necessary to keep them the same length after splitting. This implementation is much simpler and can be done in a non-allocating way with just `reshape` and `view` on a 3d array. This will be part of https://github.com/TuringLang/MCMCDiagnosticTools.jl/issues/22.

Reply:
The existing splitting functionality `copy_split!` will go away.